### PR TITLE
TST: Update scipy.fft real transform tests

### PR DESCRIPTION
**Pull request by peterbell10**
#### What does this implement/fix?
This is a general cleanup of the tests for `dct` and `dst` in `scipy.fft`. Specific improvements include:
- using pytest parametrized tests instead of seperate classes
- grouping tolerances together into a single dict
- adding a parametrization for `np.longfloat` to all transforms
- tightening test tolerances

#### Additional information
I was also planning to address @larsoner's comment (https://github.com/scipy/scipy/pull/10493#discussion_r305904163) about changing the `FFTWDATA` globals into a fixture that implicitly parametrizes the test with the different sizes but ran into issues. pytest seems to require that test parametrization be known during collection which means that the data file needs to have already been loaded.

So, I think that a fixture can only be used if the tests go back to explicitly iterating over the different sizes. i.e. instead of
```python
def test(fftwdata_size):
	# test one size
```
it would have to be
```python
def test(fftwdata_sizes):
	for size in fftwdata_sizes:
        # test one size
```

Which of the two would be preferred and is there another way around the collection issue?

### CODE REVIEW

**Comment by larsoner** [comment]
@peterbell10 if we encode the number of sizes `n_sizes` to be tested in the file (at a slight DRY cost) that matches the number in the actual file then we can use this:

https://pytest.readthedocs.io/en/latest/example/parametrize.html#deferring-the-setup-of-parametrized-resources

You could parametrize to run `np.arange(n_sizes)` and deferred-load the actual size.

**Comment by peterbell10** [comment]
Can the `conftest.py` file be local to just the `scipy.fft` tests?

**Comment by larsoner** [comment]
I *think* you can nest `conftest.py` files, so you could add one in `scipy/fft`.

**Comment by peterbell10** [comment]
Tests are failing on 32-bit linux because the `long double` reference data isn't binary compatible. Is there an alternative file format I could use or should I just skip the test?

**Comment by larsoner** [comment]
You could write it to `.mat` with `scipy.io`. It would be good to make the tests work on 32-bit somehow.

**Comment by peterbell10** [comment]
I just checked and `savemat` is actually casting to `np.float64` before writing to the file.

**Review by larsoner — APPROVED** [review]
> I just checked and savemat is actually casting to np.float64 before writing to the file.

Bummer. I'm fine with skipping the longdouble ones on 32-bit then.

+1 for merge from me. @grlee77 do you want to have a look?

**Comment by larsoner** [comment]
(Travis failure appears to be unrelated, see #10676)

**Comment by larsoner** [comment]
I'll go ahead and merge this one. It's mostly a refactor/test modernization so shouldn't be too controversial.

**Comment by larsoner** [comment]
Thanks @peterbell10

File: scipy/fftpack/tests/gen_fftw_ref.py [hunk 1]
```diff
@@ -57,3 +59,18 @@ def gen_data(dt):
     for sz in SZ:
         d['dst_%d_%d' % (type-4, sz)] = data[type][sz]
 np.savez(filename, **d)
+
+# generate long double precision data
+data = gen_data(np.float128)
```
Comment (tylerjereddy) [inline scipy/fftpack/tests/gen_fftw_ref.py]: Portability to `ppc64le` native hardware? see #9684 after rebase

**Pull request merged**